city_scrapers/spiders/chi_state_university.py

```diff
@@ -0,0 +1,151 @@
+import re
+from calendar import month_name
+from datetime import datetime
+
+from city_scrapers_core.constants import BOARD, COMMITTEE
+from city_scrapers_core.items import Meeting
+from city_scrapers_core.spiders import CityScrapersSpider
+
+
+class ChiStateUniversitySpider(CityScrapersSpider):
+    name = "chi_state_university"
+    agency = "Chicago State University"
+    timezone = "America/Chicago"
+    start_urls = ["https://www.csu.edu/boardoftrustees/dates.htm"]
+
+    def parse(self, response):
+        """
+        `parse` should always `yield` Meeting items.
+
+        Change the `_parse_title`, `_parse_start`, etc methods to fit your scraping
+        needs.
+        """
+
+        for data in response.xpath("/html/body/div/div[4]/div/div[1]/div/div"):
+            titles = data.xpath(".//label/text()").extract()
+
+            for i, tab in enumerate(data.xpath('.//div[@class = "itd-tab"]')):
+
+                for item in tab.xpath(".//li"):
+
+                    months = {m.lower() for m in month_name[1:]}
+                    text = item.xpath(".//text()").extract()[0]
+                    monthMatch = next(
+                        (word for word in text.split() if word.lower() in months), None
+                    )
+                    if monthMatch is None:
+                        continue
+
+                    meeting = Meeting(
+                        title=titles[i],
+                        description=self._parse_description(item),
+                        classification=self._parse_classification(i),
+                        start=self._parse_start(item),
+                        end=None,
+                        all_day=False,
+                        time_notes=self._parse_time_notes(item),
+                        location=self._parse_location(item),
+                        links=self._parse_links(item),
+                        source=self._parse_source(response),
+                    )
+                    meeting["status"] = self._get_status(meeting)
+                    meeting["id"] = self._get_id(meeting)
+
+                    yield meeting
+
+    def _parse_description(self, item):
+        parts = item.xpath(".//text()").extract()
+        description = "".join([str(x) for x in parts])
+        description = (
+            description.replace("\xa0", " ")
+            .replace("\n", " ")
+            .replace("                                     ", " ")
+            .replace("\t", " ")
+            .strip()
+        )
+        return description
+
+    def _parse_classification(self, i):
+        classifications = [BOARD, BOARD, COMMITTEE]
+        return classifications[i]
+
+    def _parse_start(self, item):
+        text = item.xpath(".//text()").extract()[0]
```

Comment:
Is there a reason for creating the datetime from each component piece rather than calling `datetime.strptime` with a format (or maybe formats)? This seems like it could return false positives if the pattern changes rather than failing directly, and it looks like the dates are generally pretty standard

Reply:
Yes I wasn't very happy with this approach but it was what I settled on after battling with the formatting for a long time. I used `"%A, %B %d, %Y @ %H:%M %p"` and `"%B %d, %Y @ %H:%M %p"` for instances without the day at the beginning. But for some reason this wasn't working on any of the date and time strings. I tried slight variations as well as parsing some segments of the date and trying `datetime.strptime` on just those parts - this includes stripping whitespace and odd characters and so on but it wouldn't play ball. I've not come across such awkward behavior before so if you have any suggestions that would be useful.
I went with this approach also because there were instances of dates without time or minutes and so on, so as a catch all I tried to account for any kind of supplied date as much as I could.

Reply:
Hmm thanks for clarifying! We've generally been removing the day string at the beginning because it doesn't add necessary info (and sometimes they're wrong), so that could be one place to start. It can also help to normalize the whitespace with something like:

```python
date_str = re.sub(r"\s+", " ", date_str)
```

That way you don't have to hunt down any specific hidden whitespace characters. And one other thing that has helped is removing commas and periods since they're typically unneeded and in many cases inconsistent. I'm not sure if `strptime` accepts "a.m." vs "am" either. If it's alright with you it would be great to give that another shot, but let me know if you run into specific edge cases!